ambari-web/app/controllers/wizard/customProductRepos_controller.js

```diff
@@ -46,115 +44,116 @@ App.WizardCustomProductReposController = App.WizardStepController.extend({
-      mpacks.pushObject(Em.Object.create({
-        id: stack.get('id'), //this is actually the stack id from App.Stack, which is actually the repository_version id in the database, which is an integer
-        name: mpack.name,
-        displayName: mpack.displayName,
-        publicUrl: mpack.publicUrl,
-        downloadUrl: mpack.downloadUrl,
-        version: mpack.version,
-        operatingSystems: stack ? stack.operatingSystems : []
-      }));
-    });
-    this.set('mpacks', mpacks);
-
-    const repos = this.get('mpacks').reduce(
-      (repos, mpack) => repos.concat(
-        mpack.get('operatingSystems').reduce(
-          (repos, os) => repos.concat(
-            os.get('repos')
+        }))
+      });
+      this.set('mpacks', mpacks);
+
+      const repos = mpacks.reduce(
+        (repos, mpack) => repos.concat(
+          mpack.get('operatingSystems').reduce(
+            (repos, os) => repos.concat(
+              os.get('repos')
+            ),
+            [])
           ),
-          [])
-        ),
-      []
-    );
-    this.set('repos', repos);
+        []
+      );
+      this.set('repos', repos);
 
-    const uniqueOperatingSystems = {};
-    mpacks.forEach(mpack => {
-      mpack.get('operatingSystems').forEach(os => {
-        const osType = os.get('type');
-        uniqueOperatingSystems[osType]
-          ? uniqueOperatingSystems[osType].mpacks.pushObject(mpack)
-          : uniqueOperatingSystems[osType] = {
-              selected: os.get('selected'),
-              mpacks: [mpack]
-            };
-      })
-    });
-    
-    const operatingSystems = [];
-    for (let osType in uniqueOperatingSystems) {
-      operatingSystems.pushObject(Em.Object.create({
-        type: osType,
-        selected: uniqueOperatingSystems[osType].selected,
-        mpacks: uniqueOperatingSystems[osType].mpacks
-      }))
-    }
-    operatingSystems.sort((a, b) => a.get('type').localeCompare(b.get('type')));
-    this.set('operatingSystems', operatingSystems);
+      const uniqueOperatingSystems = {};
+      mpacks.forEach(mpack => {
+        mpack.get('operatingSystems').forEach(os => {
+          const osType = os.get('type');
+          uniqueOperatingSystems[osType]
+            ? uniqueOperatingSystems[osType].mpacks.pushObject(mpack)
+            : uniqueOperatingSystems[osType] = {
+                selected: os.get('selected'),
+                mpacks: [mpack]
+              };
+        })
+      });
+      
+      const operatingSystems = [];
+      for (let osType in uniqueOperatingSystems) {
+        operatingSystems.pushObject(Em.Object.create({
+          type: osType,
+          selected: uniqueOperatingSystems[osType].selected,
+          mpacks: uniqueOperatingSystems[osType].mpacks
+        }))
+      }
+      operatingSystems.sort((a, b) => a.get('type').localeCompare(b.get('type')));
+      this.set('operatingSystems', operatingSystems);
+    },
+      
+    this.wizardController.addErrors);
```

Comment:
This looks wrong, I think it should be this.wizardController.addErrors();

Reply:
Actually, it should be this.get('wizardController').addErrors, which I fixed. I am passing the function itself to be the callback for the promise; I do not want to pass the result of the function.